test/controllers/shared/basic_rest_response_test.rb

```diff
@@ -36,5 +36,108 @@ def basic_edit_test(object_found = nil)
         end
       end
     end
+
+    def basic_pagination_per_page_test
+      context 'GET #index' do
+        entries_per_page = Setting[:entries_per_page] ? Setting[:entries_per_page] : 5
+        setup do
+          factory_name =  get_factory_name(@controller.resource_class)
+          create_model_list(factory_name, entries_per_page + 2)
+          get :index, {per_page: entries_per_page + 1}, set_session_user
+          assert_response :success
+        end
+
+        test 'should render correct per_page value' do
+          per_page_results = css_select('.pagination-pf-items-current')
+          assert_equal "1-#{entries_per_page + 1}", per_page_results.first.content.squish
+        end
+
+        test 'should render per page dropdown with correct values' do
+          assert_select "[id='per_page']" do
+            assert_select "option[selected='selected'][value='#{entries_per_page + 1}']"
+            assert_select "option[value='5']"
+            assert_select "option[value='10']"
+            assert_select "option[value='15']"
+            assert_select "option[value='25']"
+            assert_select "option[value='50']"
+          end
+        end
+
+        test 'should display pagination buttons' do
+          get :index, {per_page: entries_per_page + 1}, set_session_user
+          assert_response :success
+          assert_select "[id=pagination]" do
+            assert_includes @response.body.squish, "pagination pagination-pf-back"
+            assert_includes @response.body.squish, "pagination pagination-pf-forward"
+          end
+        end
+
+        test 'sort links should include per page param' do
+          sort_links = css_select('thead a')
+          sort_links.each do |link|
+            assert_includes link['href'], "per_page=#{entries_per_page + 1}"
+          end
+        end
+      end
+    end
+
+    def basic_pagination_rendered_test
+      context 'GET #index' do
+        old = Setting[:entries_per_page]
```

Comment:
This should go inside the setup block, and use a local variable (`@old`) so it is accessible in the teardown block.